examples/vrf/main.tf

```diff
@@ -75,4 +75,38 @@ resource "dcnm_vrf" "second" {
     loopback_id   = 70
     loopback_ipv4 = "1.2.3.4"
   }
-}
\ No newline at end of file
+}
+
+resource "dcnm_vrf" "third" {
+  fabric_name        = "fab3"
+  name               = "check3"
+  segment_id         = 50050
+  template           = "Template_Universal"
+  extension_template = "Template_Extension_Universal"
+  template_props = {
+    "advertiseDefaultRouteFlag" : "true"
+    "vrfVlanId" : "123"
+    "isRPExternal" : "false"
+    "vrfDescription" : ""
+    "maxBgpPaths" : "6"
+    "maxIbgpPaths" : "6"
+    "borderMaxBgpPaths" : "16"
+    "ipv6LinkLocalFlag" : "false"
+    "vrfRouteMap" : "RM_ADVERTISE_CONNECTED_SVI"
+    "ENABLE_NETFLOW" : "false"
+    "bgpPassword" : ""
+    "mtu" : "9000"
+    "multicastGroup" : ""
+    "isRPAbsent" : "false"
+    "advertiseHostRouteFlag" : "true"
+    "vrfVlanName" : ""
+    "trmEnabled" : "false"
+    "asn" : "4201020601"
+    "vrfIntfDescription" : ""
+    "vrfSegmentId" : "50050"
+    "vrfName" : "check3"
```

Comment:
Even if the VRF Name and the VRF Segment ID are inputs of the resource, the values must be set inside the template properties as well. Otherwise Terraform will try to update does values in a subsequent `terraform apply`. This is because the Read Operataion identifies those keys but the Schema does not have them

Reply:
The same applies for the Network Resource in the PR https://github.com/CiscoDevNet/terraform-provider-dcnm/pull/121

Reply:
Got it. This PR is working for me. Tested it out with different scenarios.